pkg/buildcontext/azureblob.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package buildcontext
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"path/filepath"
+
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"github.com/GoogleContainerTools/kaniko/pkg/constants"
+	"github.com/GoogleContainerTools/kaniko/pkg/util"
+)
+
+// GCS struct for Google Cloud Storage processing
+type AzureBlob struct {
+	context string
+}
+
+func (b *AzureBlob) UnpackTarFromBuildContext() (string, error) {
```

Comment:
Why not NewAzureBlobFromContext()? It's more inline with other go packages and then your struct is actually a struct of those useful strings, instead of just being a wrapper for a string.

Reply:
We will keep this for now to keep the code structure align with how google and s3 implemented right now https://github.com/yangtaokm/kaniko/blob/e9af4cfbf6ae6ff5a38cbf0e1011a4fa168a5bc2/pkg/buildcontext/gcs.go#L31-L33